drivers/serial/Kconfig.bt

```diff
@@ -3,7 +3,7 @@
 
 config UART_BT
 	bool "UART over NUS Bluetooth LE"
-	depends on BT_NUS
+	depends on BT_NUS_ZEPHYR
 	depends on DT_HAS_ZEPHYR_NUS_UART_ENABLED
```

Comment:
Looks a bit odd to have `BT_NUS_ZEPHYR` and `HAS_ZEPHYR_NUS` right next to each other. Should that be streamlines to be `NUS_ZEPHYR` or `ZEPHYR_NUS` in all cases?

Reply:
`DT_HAS_ZEPHYR_NUS_UART_ENABLED` comes from the dts-node compatible with `zephyr,nus-uart`. `BT_NUS_ZEPHYR` comes from what was originally `BT_NUS`.

I wouldn't suggest dropping the `BT_` from the Kconfig, as it breaks the name-spacing pattern used for the other Bluetooth Services.

Having said that, I don't love the `BT_NUS_ZEPHYR` either. If it looks confusing, I suggest coming up with another prefix-ending in the form `BT_NUS_XXX`.

Some alternatives:
- BT_NUS_UPSTREAM.
- BT_NUS_Z.
- ???

LMK if you have any suggestions!

Reply:
> I wouldn't suggest dropping the BT_ from the Kconfig, as it breaks the name-spacing pattern used for the other Bluetooth Services.

Sorry, I didn't meant to suggest that either. It was exclusively about the order of `NUS` and `ZEPHYR` where now we have both `NUS_ZEPHYR` and `ZEPHYR_NUS`, and felt it made more sense to only have it in 1 order. 

> LMK if you have any suggestions!
Someone mentioned that the `N` in `NUS` is for Nordic, so I (half-jokingly) suggest `ZUS` (Zephyr UART Service) :D But I also know that will likely never be approved (and for good reasons).

Reply:
`BT_🪁_NUS`

Reply:
@Thalley AFAIK - `DT_HAS_ZEPHYR_NUS_UART_ENABLED` is a direct expansion from the dts compatible property:`zephyr,nus-uart`, which usually follows `vendor,value`. I think that's harder to change than the Kconfig. Maybe I'm missing something, though. Some other examples of this: `zephyr,cdc_acm_uart`, `nordic,nrf-usbd` or `nordic,nrf-spim`.

Per @alwa-nordic's vote: I'll change it to `BT_ZEPHYR_NUS`.

Reply:
Even though there's no consensus on the name-spacing, it seems like a one-off situation where a service is hosted in more than one repository. The logical option would be just `BT_NUS` but it's taken.

Changed Kconfigs to `BT_ZEPHYR_NUS`, unless anyone has strong feelings against it.